src/installer.ts

```diff
@@ -1,137 +1,166 @@
-  }
-  fs.copyFileSync(
-    path.join(toolPath, denoBinaryName),
-    path.join(denoBin, denoBinaryName)
-  );
-  fs.chmodSync(path.join(denoBin, denoBinaryName), 0o755);
-
-  //
-  // prepend the tools path. instructs the agent to prepend for future tasks
-  core.addPath(denoBin);
-}
-
-function denoBinPath(): string {
-  switch (os.platform()) {
-    case "darwin":
-      return path.join(process.env.HOME || "", ".deno", "bin");
-    case "linux":
-      return path.join(process.env.HOME || "", ".deno", "bin");
-    case "win32":
-      return path.join(process.env.USERPROFILE || "", ".deno", "bin");
-    default:
-      throw Error("Invalid platform");
-  }
-}
-
-async function acquireDeno(version: string): Promise<string> {
-  //
-  // Download - a tool installer intimately knows how to get the tool (and construct urls)
-  //
-  let platform: "osx" | "linux" | "win";
-  let extension: "gz" | "zip";
-  let executableExtension = "";
-
-  switch (os.platform()) {
-    case "darwin":
-      platform = "osx";
-      extension = "gz";
-      break;
-    case "linux":
-      platform = "linux";
-      extension = "gz";
-      break;
-    case "win32":
-      platform = "win";
-      extension = "zip";
-      executableExtension = ".exe";
-      break;
-    default:
-      throw Error("Invalid platform");
-  }
-
-  core.debug(
-    `Trying to install for platform ${platform} with extension ${extension}`
-  );
-
-  let toolName = `deno_${platform}_x64`;
-  let downloadUrl = `https://github.com/denoland/deno/releases/download/${version}/${toolName}.${extension}`;
-  let downloadPath: string;
-
-  try {
-    downloadPath = await tc.downloadTool(downloadUrl);
-  } catch (err) {
-    throw err;
-  }
-
-  core.debug(`Downloaded ${downloadUrl} to ${downloadPath}`);
-
-  //
-  // Extract
-  //
-  let extPath: string;
-  if (!fs.existsSync(tempDirectory)) {
-    fs.mkdirSync(tempDirectory, {
-      recursive: true
-    });
-  }
-  if (extension == "zip") {
-    extPath = await tc.extractZip(downloadPath, tempDirectory);
-    extPath = tempDirectory;
-    toolName = "deno" + executableExtension;
-  } else if (extension == "gz") {
-    const buffer = fs.readFileSync(downloadPath);
-    const bin = await ungzip(buffer);
-    extPath = tempDirectory;
-    fs.writeFileSync(path.join(extPath, toolName), bin);
-  } else {
-    throw Error("Unknown extension");
-  }
-  core.debug(`Extracted archive to ${extPath}`);
-
-  //
-  // Install into the local tool cache - deno extracts a file that matches the fileName downloaded
-  //
-  let tool = path.join(extPath, toolName);
-  core.debug(`Cache file ${tool} into toolcache`);
-  return await tc.cacheFile(
-    tool,
-    "deno" + executableExtension,
-    "deno",
-    version
-  );
-}
+import * as os from "os";
+import * as path from "path";
+
+// On load grab temp directory and cache directory and remove them from env (currently don't want to expose this)
+let tempDirectory: string = process.env["RUNNER_TEMP"] || "";

```

Comment:
Is there any reason to put this before `import`?

Reply:
https://github.com/actions/toolkit/blob/master/packages/tool-cache/src/tool-cache.ts#L24
tool-cache gets tempDir from env, so we need to set env before import